### DISCUSSION TITLE
DAI 2023-2024 - Java IOs benchmarking - Simon Guggisberg

### DISCUSSION THREAD

**Discussion by GuggisbergSimon** [heig-vd-dai-course/heig-vd-dai-course#81] [Show and tell]
https://github.com/GuggisbergSimon/heig-vd-dai-course-java-ios-practical-content
The results and discussion of the points are listed in the readme

**Comment by ludelafo** [comment]
Good! Thank you.

Regarding your comment _"Methodology ? Not sure what you're referring to."_, I wanted to make you aware on how measuring the time execution of your code is an important step (a scientific methodology) to ensure your code is optimized and well fit for a given use-case.

Using tangible metrics, you can always reuse this methodology in another context to validate if the changes you made to your application are still relevant for your use-case or not.

**Edit**: You could add the `.txt` and `.bin` files to the gitignore file to avoid them on the Git repository.

**Edit 2**: Do not forget to set the character encoding when processing text files! The `FileReader` and `FileWriter` classes can use a charset: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/FileReader.html and https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/FileWriter.html

**Reply by GuggisbergSimon** [reply]
Hello, Pushed those changes earlier, thank you for the feedback.
Cheers

**Reply by ludelafo** [reply]
You're welcome!
